Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,399 @@
+            {
+                curState.RawPosition = m_PreviousCameraPosition = pathSpline[0].Position;
+                curState.RawOrientation = m_PreviousOrientation = pathSpline[0].Rotation;
+                return;
+            }
+            
+            m_PathPosition = 
+                SplineUtility.ConvertIndexUnit(pathSpline, m_PathPosition, m_PositionUnits, PathIndexUnit.Normalized);
+            
+            // Init previous frame state info
+            if (deltaTime < 0 || !VirtualCamera.PreviousStateIsValid)
+            {
+                m_PreviousNormalizedPathPosition = m_PathPosition;
+                m_PreviousCameraPosition = curState.RawPosition;
+                m_PreviousOrientation = curState.RawOrientation;
+            }
+
+            // Get the new ideal path base position
+            if (m_AutoDolly.m_Enabled && FollowTarget != null)
+            {
+                // convert follow target into spline local space, because SplineUtility works in spline local space
+                SplineUtility.GetNearestPoint(pathSpline, 
+                    m_Track.transform.InverseTransformPoint(FollowTargetPosition), out _, out m_PathPosition, 
+                    m_AutoDolly.m_SearchResolution, m_AutoDolly.m_SearchIteration);
+                // Apply the path position offset
+                m_PathPosition += m_AutoDolly.m_PositionOffset;
+            }
+            float newPathPosition = m_PathPosition;
+
+            if (deltaTime >= 0 && VirtualCamera.PreviousStateIsValid)
+            {
+                float maxUnit = 1; // we are always using normalized unit [0-1]
+                {
+                    float prev = m_PreviousNormalizedPathPosition;
+                    float next = newPathPosition;
+                    if (pathSpline.Closed && Mathf.Abs(next - prev) > maxUnit / 2)
+                    {
+                        if (next > prev)
+                            prev += maxUnit;
+                        else
+                            prev -= maxUnit;
+                    }
+                    m_PreviousNormalizedPathPosition = prev;
+                    newPathPosition = next;
+                }
+
+                // Apply damping along the path direction
+                float offset = m_PreviousNormalizedPathPosition - newPathPosition;
+                offset = Damper.Damp(offset, m_Damping.z, deltaTime);
+                newPathPosition = m_PreviousNormalizedPathPosition - offset;
+            }
+            m_PreviousNormalizedPathPosition = newPathPosition;
+            m_Track.Evaluate(newPathPosition, 
+                out var localPosition, out var localTangent, out var localUp);
+            Vector3 newCameraPos = localPosition;
+            var newPathOrientation = 
+                Vector3.SqrMagnitude(localTangent) == 0 || Vector3.SqrMagnitude(localUp) == 0 ? 
+                    Quaternion.identity : Quaternion.LookRotation(localTangent, localUp);
```

Comment:
Is it possible that localTangent and localUp are parallel?  What happens in that case?  Try making a path that goes straight up and adding roll to it.  Does it work as expected?